0000-settlement-engines/0000-settlement-engines.md

```diff
@@ -0,0 +1,395 @@
+
+#### Request
+
+```http
+PUT /accounts/:id HTTP/1.1
+```
+
+#### Response
+
+```http
+HTTP/1.1 200 OK
+```
+
+### Delete an account
+
+Instructs the settlement engine that an account was deleted.
+
+#### Request
+
+```http
+DELETE /accounts/:id HTTP/1.1
+```
+
+#### Response
+
+```http
+HTTP/1.1 200 OK
+```
+
+### Perform outgoing settlement
+
+Asynchronously send an outgoing settlement. The accounting system sends this request and [accounts for outgoing settlements](#account-for-outgoing-settlements).
+
+#### Request
+
+```http
+POST /accounts/:id/settlements HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+Idempotency-Key: <key>
+```
+
+> **[`Quantity`](#quantity-json-type)** to settle
+
+#### Response
+
+```http
+HTTP/1.1 202 ACCEPTED
+Content-Type: application/json
+```
+
+> **[`Quantity`](#quantity-json-type)** enqueued to settle, which MUST be less than or equal to the quantity in the original request
+
+### Handle incoming message
+
+Process and respond to an incoming message from the peer's settlement engine. The connector sends this request when it receives it an incoming settlement message from the peer, and returns the response message back to the peer.
+
+#### Request
+
+```http
+POST /accounts/:id/messages HTTP/1.1
+Accept: application/octet-stream
+Content-Type: application/octet-stream
+```
+
+> _&lt;raw bytes of message&gt;_
+
+#### Response
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/octet-stream
+```
+
+> _&lt;raw bytes of response&gt;_
+
+## Accounting System HTTP API
+
+_Settlement engine &rarr; connector_
+
+HTTP/2 is RECOMMENDED for performance reasons, although HTTP/1.1 MAY also be used. Implementations SHOULD support HTTP version negotiation via Application Protocol Negotiation (ALPN).
+
+### Credit incoming settlement
+
+[Account for an incoming settlement](#account-for-incoming-settlements). The settlement engine sends this request as it receives incoming settlements in in order to ensure [settlement symmetry](#settlement-symmetry). If the accounting system responds that it only credited a partial amount, the settlement engine [tracks the uncredited amounts](#track-uncredited-incoming-settlements).
+
+#### Request
+
+```http
+POST /accounts/:id/settlements HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+Idempotency-Key: <key>
+```
+
+> **[`Quantity`](#quantity-json-type)** to be credited to the account as an incoming settlement
+
+#### Response
+
+```http
+HTTP/1.1 201 CREATED
+Content-Type: application/json
+```
+
+> **[`Quantity`](#quantity-json-type)** credited to the account as an incoming settlement, which MUST  be less than or equal to the quantity in the original request
+
+### Send outgoing message
+
+Send the message to the given peer's settlement engine and return its response. The connector handles [proxying the message](#exchanging-messages) through the peer's connector.
```

Comment:
Maybe add some context on why this call does not need to be idempotent on the connector, since we assume that the engine's API handles the idempotency?